sdk/deviceupdate/iot-device-update-rest/samples-dev/getDevice.ts

```diff
@@ -19,17 +19,60 @@ const instanceId = process.env["INSTANCE_ID"] || "";
 
 async function main() {
   console.log("== Get device ==");
-  const device = process.env["DEVICEUPDATE_DEVICE"] || "";
+  const groupId = process.env["DEVICEUPDATE_DEVICE_GROUP"] || "";
 
   const credentials = new DefaultAzureCredential();
 
   const client = DeviceUpdate(endpoint, credentials);
 
-  const result = await client
-    .path("/deviceUpdate/{instanceId}/management/devices/{deviceId}", instanceId, device)
+  console.log("Get various device management information from Device Update for IoT Hub...");
+
+  console.log("\nDevices:");
+  const devicesResult = await client
```

Comment:
this seems to be a pageable operation. You should be able to use the `paginate` helper. You can import it this way

`import DeviceUpdate, { isUnexpected, paginate } from "@azure-rest/iot-device-update";`

This will give you an iterator which can be used like this:

```ts
const devicesResult = await client
.path("/deviceUpdate/{instanceId}/management/devices", instanceId)
.get();

const devices = paginate(client, devicesResult);

for await (const device of devices) {
   console.log(device.deviceId)
}
```

There are other pageable operations that can use this pattern

Reply:
Fixed.